File: app/src/main/java/com/recite/zz/kotlin/view/DailyView.java
package com.recite.zz.kotlin.view;

import android.content.Context;
import android.util.AttributeSet;
import android.widget.LinearLayout;
import android.widget.TextView;

import com.recite.zz.kotlin.R;
import com.recite.zz.kotlin.repository.data.DailySentence;

import java.util.List;


/**
 * Created with Android Studio.
 * User: Michael
 * Date: 2015/11/19
 * Time: 23:07
 */
public class DailyView extends LinearLayout {
    private TextView mDailySentence;
    private TextView mDailyTrans;
    private TextView mDailyNote;
    private TextView mDailyTitle;
    private List<DailySentence> mAllSentences;
    private int mPosition;

    public DailyView(Context context) {
        super(context);
    }

    public DailyView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }


    @Override
    protected void onFinishInflate() {
        super.onFinishInflate();
        mDailyTitle = (TextView) findViewById(R.id.dict_daily_title);
        mDailySentence = (TextView) findViewById(R.id.dict_daily_sentence);
        mDailyTrans = (TextView) findViewById(R.id.dict_daily_sentence_trans);
        mDailyNote = (TextView) findViewById(R.id.dict_daily_sentence_note);

    }

    private void bindView(DailySentence sentence) {
        mDailyTitle.setText(getContext().getString(R.string.ciba_daily) + " " + sentence.getDateline());
        mDailySentence.setText(sentence.getContent());
        mDailyTrans.setText(sentence.getNote());
        mDailyNote.setText(sentence.getTranslation());
    }

    public int getHeightNeeded(){
        return getPaddingTop() + getPaddingBottom()+
                mDailyTitle.getLineCount()*mDailyTitle.getLineHeight()+
                mDailySentence.getLineCount()*mDailySentence.getLineHeight()+
                mDailyTrans.getLineCount()*mDailyTrans.getLineHeight()+
                mDailyNote.getLineCount()*mDailyNote.getLineHeight()+
                mDailyTitle.getPaddingTop()+
                mDailyTitle.getPaddingBottom()+50
                ;


    }

    public void updateSentence(List<DailySentence> sentences, int position) {
        mAllSentences = sentences;
        mPosition = position;
        DailySentence sentence = mAllSentences.get(position);
        if (sentence != null) {
            bindView(sentence);
        }
    }

    public void showNext() {
        mPosition++;
        mPosition = mPosition % mAllSentences.size();
        DailySentence sentence = mAllSentences.get(mPosition);
        if (sentence != null) {
            bindView(sentence);
        }
    }

    public boolean shouldShow(){
        return mAllSentences!=null && !mAllSentences.isEmpty();
    }
}

File: app/src/main/java/com/recite/zz/kotlin/view/DailyView.kt
package com.recite.zz.kotlin.view

import android.content.Context
import android.util.AttributeSet
import android.widget.ImageView
import android.widget.LinearLayout
import android.widget.TextView

import com.recite.zz.kotlin.R
import com.recite.zz.kotlin.ext.loadUrl
import com.recite.zz.kotlin.ext.screenWidth
import com.recite.zz.kotlin.repository.data.DailySentence
import kotlinx.android.synthetic.main.dailyview_layout.view.*


/**
 * Created with Android Studio.
 * User: Michael
 * Date: 2015/11/19
 * Time: 23:07
 */
class DailyView : LinearLayout {
    private var mHeadImg: ImageView? = null
    private var mDailySentence: TextView? = null
    private var mDailyTrans: TextView? = null
    private var mDailyNote: TextView? = null
    private var mDailyTitle: TextView? = null
    private var mAllSentences: List<DailySentence>? = null
    private var mPosition: Int = 0

    val heightNeeded: Int
        get() = paddingTop + paddingBottom +
                mDailyTitle!!.lineCount * mDailyTitle!!.lineHeight +
                mDailySentence!!.lineCount * mDailySentence!!.lineHeight +
                mDailyTrans!!.lineCount * mDailyTrans!!.lineHeight +
                mDailyNote!!.lineCount * mDailyNote!!.lineHeight +
                mDailyTitle!!.paddingTop +
                mDailyTitle!!.paddingBottom + 50 + resources.screenWidth * 260/ 160

    constructor(context: Context) : super(context)

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs)


    override fun onFinishInflate() {
        super.onFinishInflate()
        mHeadImg = findViewById(R.id.headImg)
        mDailyTitle = findViewById(R.id.dict_daily_title)
        mDailySentence = findViewById(R.id.dict_daily_sentence)
        mDailyTrans = findViewById(R.id.dict_daily_sentence_trans)
        mDailyNote = findViewById(R.id.dict_daily_sentence_note)

    }

    private fun bindView(sentence: DailySentence) {
        headImg.loadUrl(sentence.picture)

        mDailyTitle!!.text = context.getString(R.string.ciba_daily) + " " + sentence.dateline
        mDailySentence!!.text = sentence.content
        mDailyTrans!!.text = sentence.note
        mDailyNote!!.text = sentence.translation
    }

    fun updateSentence(sentences: List<DailySentence>, position: Int) {
        mAllSentences = sentences
        mPosition = position
        val sentence = mAllSentences!![position]
        if (sentence != null) {
            bindView(sentence)
        }
    }

    fun showNext() {
        mPosition++
        mPosition = mPosition % mAllSentences!!.size
        val sentence = mAllSentences!![mPosition]
        if (sentence != null) {
            bindView(sentence)
        }
    }

    fun shouldShow(): Boolean {
        return mAllSentences != null && !mAllSentences!!.isEmpty()
    }
}
